### DISCUSSION TITLE
Discussion category forms not recognized when category name contains the variation selector-16 character

### DISCUSSION THREAD

**Discussion by AAriam** [community/community#79364] [Discussions]
### Select Topic Area

Bug

### Body

When a discussion category is created with a name that contains an emoji with variation selector-16 (for example, the ⌨️ emoji), the slug name for that category will contain the URL-encoded variation-selector-16 character. For example, if the discussion name is "App🐞Bug Report⌨️CLI", then the generated slug will be `app-bug-report-%EF%B8%8Fcli`.

In such a case, the corresponding form for that discussion category will not be recognized, i.e., although the form exists in the `.github/DISCUSSION_TEMPLATE` directory, when a new discussion is being created, the form is not loaded.

For the filename of the discussion form, three possible variants have been tested:
1) Filename exactly matching the URL-encoded slug: `app-bug-report-%EF%B8%8Fcli.yaml`
2) Filename exactly matching the URL-decoded slug: `app-bug-report-️cli.yaml` (notice that while not visible, the variation-selector-16 character is contained within the name)
3) Filename matching the slug without the variation-selector-16 character: `app-bug-report-cli.yaml`

None of these name variations work, meaning that no form can be used for a discussion category that contains the variation-selector-16 character in its name.

**Comment by Mustafa-Hassan2001** [comment]
-> Try adjusting the filename of the discussion form to include the URL-encoded variation-selector-16 character. Based on your example, the filename should be:
```
app-bug-report-%EF%B8%8Fcli.yaml

``` 
Ensure that this filename exactly matches the URL-encoded slug. Sometimes, case sensitivity or slight variations in the filename can cause issues.

-> Verify that the encoding of the variation-selector-16 character is consistent throughout your system. Ensure that it is consistently URL-encoded in both the category name and the form filename. Inconsistencies in encoding can lead to recognition issues.


-> If the issue persists, try adjusting the category slug without the variation-selector-16 character. Modify the category name to remove or replace the emoji with a different one that does not contain the variation-selector-16 character. Update the category slug accordingly, and check if the form is recognized.

->Check your configuration files and make sure there are no conflicting settings or restrictions that might prevent the correct loading of discussion forms. Ensure that the configuration is aligned with the expected behavior for handling special characters.

-> If you are using GitHub Actions or any other automation workflow, review the workflow file to confirm that it correctly handles special characters in discussion category names. Adjust any scripts or configurations related to discussion creation.

-> After making changes, thoroughly test the process by creating a new discussion with a category name containing the variation-selector-16 character. Verify that the correct form is loaded, and the discussion is created successfully.
